src/vs/workbench/common/editor/editorGroupModel.ts

```diff
@@ -570,22 +570,22 @@ export class EditorGroupModel extends Disposable implements IEditorGroupModel {
 					}
 				}
 
-				const newSelection = this.selection.filter(selected => !selected.matches(newActive) && !selected.matches(editor));
-				this.doSetSelection(newActive, this.editors.indexOf(newActive), newSelection);
+				const newInactiveSelection = this.selection.filter(selected => !selected.matches(newActive) && !selected.matches(editor));
+				this.doSetSelection(newActive, this.editors.indexOf(newActive), newInactiveSelection);
 			}
 
 			// One Editor
 			else {
-				this.active = null;
+				this.active = null; // TODO this potentially changes selection without a related event. and is it expected to change the selection?
```

Comment:
@benibenj this changes selection without event

Reply:
Yes selection needs be removed as there is no open editor left. There was never an active editor change event so there also doesn't need one for selection.